File: app/src/main/java/com/challenge/fidoreader/fagment/FingerItem.java
package com.challenge.fidoreader.fagment;

import android.os.Parcel;
import android.os.Parcelable;
import android.util.Log;

import com.challenge.fidoreader.R;

public class FingerItem implements Parcelable {
    String templateID;
    String fingerName;
    int resid;

    public FingerItem(String templateID, String fingerName, int resid) {
        Log.v("FingerItem",fingerName);
        this.templateID = templateID;
        this.fingerName = fingerName;
        this.resid = R.drawable.fingerprint;
    }

    public FingerItem(String fingerName) {
        Log.v("FingerItem",fingerName);
        this.fingerName = fingerName;
    }


    protected FingerItem(Parcel in) {
        templateID = in.readString();
        fingerName = in.readString();
        resid = in.readInt();
    }

    public static final Creator<FingerItem> CREATOR = new Creator<FingerItem>() {
        @Override
        public FingerItem createFromParcel(Parcel in) {
            return new FingerItem(in);
        }

        @Override
        public FingerItem[] newArray(int size) {
            return new FingerItem[size];
        }
    };

    public String getFingerName() {
        return fingerName;
    }

    public void setFingerName(String fingerName) {
        this.fingerName = fingerName;
    }

    public int getResid() {
        return resid;
    }

    public void setResid(int resid) {
        this.resid = resid;
    }

    public String getTemplateID() {
        return templateID;
    }

    public void setTemplateID(String templateID) {
        this.templateID = templateID;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(templateID);
        dest.writeString(fingerName);
        dest.writeInt(resid);
    }
}

File: app/src/main/java/com/challenge/fidoreader/fagment/data/FingerItem.kt
package com.challenge.fidoreader.fagment.data


import android.os.Parcel
import android.os.Parcelable
import android.util.Log
import com.challenge.fidoreader.R


class FingerItem() : Parcelable {
    lateinit var templateID: String
    lateinit var fingerName: String
    var resid = 0

    constructor(templateID: String, fingerName: String, resid: Int) : this() {
        Log.v("FingerItem", fingerName)
        this.templateID = templateID
        this.fingerName = fingerName
        this.resid = R.drawable.fingerprint
    }

    constructor(fingerName: String) : this() {
        Log.v("FingerItem", fingerName)
        this.fingerName = fingerName
    }

    constructor(parcel: Parcel) : this() {
        templateID = parcel.readString().toString()
        fingerName = parcel.readString().toString()
        resid = parcel.readInt()
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(templateID)
        dest.writeString(fingerName)
        dest.writeInt(resid)
    }

    override fun describeContents(): Int {
        return 0
    }


    companion object CREATOR : Parcelable.Creator<FingerItem> {
        override fun createFromParcel(parcel: Parcel): FingerItem {
            return FingerItem(parcel)
        }

        override fun newArray(size: Int): Array<FingerItem?> {
            return arrayOfNulls(size)
        }
    }
}